icu4c/source/tools/genrb/reslist.h

```diff
@@ -107,7 +109,8 @@ struct SRBRoot {
   KeyMapEntry *fKeyMap;
   int32_t fKeysBottom, fKeysTop;
   int32_t fKeysCapacity;
-  int32_t fKeysCount;
+  int32_t fKeysTotalCount;
```

Comment:
Why can you not keep a single fKeysCount and just change it in compactKeys()?

Reply:
I made the two different fields yesterday to help myself reason about the correctness of the code.  Now that I look back I've convinced myself that you can modify the old value without side-effects.  Fixed.